### PR TITLE
Libp2p and overflow patches

### PR DESCRIPTION
**Pull request by rob-maron**

### CODE REVIEW

File: Cargo.toml
Comment (jbearer) [inline Cargo.toml:158]: Can we remove the `patch` before merging? Just actually do the HotShot update in `dependencies`?

Comment (rob-maron) [reply Cargo.toml:158]: I can try yeah

Comment (rob-maron) [reply Cargo.toml:158]: I'm getting compile errors 😢

Comment (jbearer) [reply Cargo.toml:158]: Oh right because without the `patch` we'd have to go update the HotShot version in all the dependencies

Comment (jbearer) [reply Cargo.toml:158]: I suppose we can merge with the patch to unblock staging testing for the next release, but we should quickly go back and make this an official hotshot update

**Review by jbearer — APPROVED** [review]

File: Cargo.toml
Comment (rob-maron) [reply Cargo.toml:158]: Yeah it is already on HS `main`, but I have to update the weekly release

**Pull request merged**
